internal/hcsoci/resources_wcow.go

```diff
@@ -131,13 +131,20 @@ func allocateWindowsResources(ctx context.Context, coi *createOptionsInternal, r
 		// Only need to create a CCG instance for v2 containers
 		if schemaversion.IsV21(coi.actualSchemaVersion) {
 			hypervisorIsolated := coi.HostingSystem != nil
-			ccgState, ccgInstance, err := CreateCredentialGuard(ctx, coi.actualID, cs, hypervisorIsolated)
+			ccgInstance, ccgResource, err := CreateCredentialGuard(ctx, coi.actualID, cs, hypervisorIsolated)
 			if err != nil {
 				return err
 			}
-			coi.ccgState = ccgState
-			r.resources = append(r.resources, ccgInstance)
-			//TODO dcantah: If/when dynamic service table entries is supported register the RpcEndpoint with hvsocket here
+			coi.ccgState = ccgInstance.CredentialGuard
+			r.resources = append(r.resources, ccgResource)
+			if hypervisorIsolated {
+				// If hypervisor isolated we need to update the UVMs service table
+				// on the host so when we launch the container it can connect.
```

Comment:
Could you elaborate more on this?

Reply:
Anything in particular or just an overview of why this is needed?

Reply:
The comment here doesn't really explain _why_ we need to update the service table (or what that even means). I would suggest something to the effect of:

> By default HVSocket won't allow something inside the VM to connect back to a process on the host. We need to update the HVSocket service table to allow a connection to the CCG instance on the host, so that GMSA can function.

Reply:
Makes sense. I've updated it locally to that effect, will be in with the other changes to support hot-remove. Thanks guys